docs/src/components/PlaygroundSettingsOutput.tsx

```diff
@@ -0,0 +1,157 @@
+      <label className={styles.outputResolutionLabel}>Output resolution:</label>
+      <div className={styles.resolutionInputFieldsContainer}>
+        <ResolutionInputField
+          id="width"
+          value={resolution.width}
+          maxValue={7682}
+          onValueChange={(width: number) => {
+            handleSettingsUpdate({
+              width: width,
+              height: resolution.height,
+            });
+          }}
+          setValidity={(widthValidity: boolean) => {
+            setResolutionInputsValidity({
+              width: widthValidity,
+              height: resolutionInputsValidity.height,
+            });
+            setValidity(widthValidity && resolutionInputsValidity.height);
+          }}
+        />
+        <span style={{ margin: 2 }}>&#215;</span>
+        <ResolutionInputField
+          id="height"
+          value={resolution.height}
+          maxValue={4320}
+          onValueChange={(height: number) => {
+            handleSettingsUpdate({
+              width: resolution.width,
+              height: height,
+            });
+          }}
+          setValidity={(heightValidity: boolean) => {
+            setResolutionInputsValidity({
+              width: resolutionInputsValidity.width,
+              height: heightValidity,
+            });
+            setValidity(resolutionInputsValidity.width && heightValidity);
+          }}
+        />
+      </div>
+    </div>
+  );
+}
+
+interface ResolutionInputFieldProps {
+  id: string;
+  value: number;
+  maxValue: number;
+  onValueChange: (number) => void;
+  setValidity: (boolean) => void;
+}
+
+function ResolutionInputField({
+  id,
+  value,
+  maxValue,
+  onValueChange,
+  setValidity,
+}: ResolutionInputFieldProps) {
+  enum ValidationResult {
+    Ok = 'Ok',
+    TooLargeError = 'TooLarge',
+    TooSmallError = 'TooSmall',
+    UnevenError = 'Uneven',
+    ParsingError = 'ParsingError',
+  }
+  function validationResultMessage(validationResult: ValidationResult, maxValue: number) {
+    if (validationResult == ValidationResult.Ok) {
+      return 'Everything is fine';
+    } else if (validationResult == ValidationResult.ParsingError) {
+      return "Value isn't a valid number";
+    } else if (validationResult == ValidationResult.TooLargeError) {
+      return `Value has to be not greater than ${maxValue}`;
+    } else if (validationResult == ValidationResult.TooSmallError) {
+      return 'Value has to be greater than 0';
+    } else if (validationResult == ValidationResult.UnevenError) {
+      return 'Value has to be even';
+    }
+  }
+
+  const [validationResult, setValidationResult] = useState<ValidationResult>(ValidationResult.Ok);
+  function setInputValidationResult(result: ValidationResult) {
+    if (result == ValidationResult.Ok) {
+      setValidity(true);
+    } else {
+      setValidity(false);
+    }
+    setValidationResult(result);
+  }
+
+  function parseAndValidate(input_string: string): number {
```

Comment:
extract validation logic to pure function and here only call that function and update state